Sources/Fuzzilli/FuzzIL/AbstractInterpreter.swift

```diff
@@ -200,36 +220,56 @@ public struct AbstractInterpreter {
         }
         return result
     }
-    
-    private mutating func executeEffects(_ instr: Instruction) {
+
+    private mutating func executeEffects(_ instr: Instruction) -> [(Variable, Type)] {
```

Comment:
You could let this function take the `typeChanges` array as `inout` parameter since it will anyway be concatenated to it